### PR TITLE
fix: add childLoggerFactory option to FastifyHttpOptions type

### PR DESCRIPTION
**Pull request by mauroaccornero**
- add childLoggerFactory option to FastifyHttpOptions type
- add test for childLoggerFactory option

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

fixes: #5664

### CODE REVIEW

File: fastify.d.ts [hunk 1]
```diff
@@ -65,7 +73,8 @@ declare namespace fastify {
     Server extends http.Server,
     Logger extends FastifyBaseLogger = FastifyBaseLogger
   > = FastifyServerOptions<Server, Logger> & {
-    http?: http.ServerOptions | null
+    http?: http.ServerOptions | null,
+    childLoggerFactory?: FastifyChildLoggerFactory
```
Comment (climba03003) [inline fastify.d.ts:77]: I believe it should be added to the `FastifyServerOptions` instead of `FastifyHttpOptions`

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: fastify.d.ts (hunk 1 above)
Comment (mauroaccornero) [reply fastify.d.ts]: You're right, thanks for the feedback!

**Review by climba03003 — CHANGES_REQUESTED** [review]
Can you add the test through `.test.d.ts`? It is used `tsd` to assert the correctness of types.

**Comment by mauroaccornero** [comment]
> Can you add the test through `.test.d.ts`? It is used `tsd` to assert the correctness of types.

Sure, thanks for the suggestion about tsd

**Review by climba03003 — APPROVED** [review]

**Comment by climba03003** [comment]
cc @fastify/typescript

**Review by gurgunday — APPROVED** [review]

**Pull request merged**